lib/llhttp/parser.js

```diff
@@ -0,0 +1,217 @@
+'use strict'
+
+/* global WebAssembly */
+
+const { resolve } = require('path')
+const { readFileSync } = require('fs')
+const constants = require('./constants')
+const { kMaxHeadersSize } = require('../core/symbols')
+const WASM_BUILD = resolve(__dirname, './llhttp.wasm')
+const bin = readFileSync(WASM_BUILD)
+const mod = new WebAssembly.Module(bin)
+
+const kOnMessageBegin = 0
+const kOnHeaders = 1
+const kOnHeadersComplete = 2
+const kOnBody = 3
+const kOnMessageComplete = 4
+
+const kPtr = Symbol('kPrt')
+const kStatusMessage = Symbol('kStatusMessage')
+const kHeadersFields = Symbol('kHeadersFields')
+const kHeadersValues = Symbol('kHeadersValues')
+const kBufferSize = Symbol('kBufferSize')
+const kHeadersTimeout = Symbol('kHeadersTimeout')
+const kBufferPtr = Symbol('kBufferPtr')
+const kMallocBuffer = Symbol('kMallocBuffer')
+const kGetHeaders = Symbol('kGetHeaders')
+const kReset = Symbol('kReset')
+const kMakeError = Symbol('kMakeError')
+
+const cstr = (ptr, len) =>
+  Buffer.from(inst.exports.memory.buffer, ptr, len).toString()
+
+/* eslint-disable camelcase */
+const wasm_on_message_begin = p => {
```

Comment:
btw was is `p` here? Could we make it an integer and that way us an array instead of map for lookups?

Reply:
Also fine as future TODO

Reply:
Yes it's an integer. It represents the pointer address of the parser wasm side, I guess.